pkg/apis/application/v1alpha1/app_catalog_entry_types.go

```diff
@@ -109,14 +111,22 @@ type AppCatalogEntrySpecChart struct {
 type AppCatalogEntrySpecRestrictions struct {
 	// ClusterSingleton is a flag for whether this app can be installed at most once per cluster. Default is false.
 	ClusterSingleton bool `json:"clusterSingleton,omitempty"`
-	// NamespaceSingleton is a flag for whether this app can be installed at most once per namespace. Default is false.
-	NamespaceSingleton bool `json:"namespaceSingleton,omitempty"`
 	// FixedNamespace is the namespace which this app must be installed in.
 	FixedNamespace string `json:"fixedNamespace,omitempty"`
 	// GpuInstances is a flag for whether this app requires GPU instances to run. Default is false.
 	GpuInstances bool `json:"gpuInstances,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +nullable
+	// CompatibleProviders is a list of provider names which this app is compatible with. Default to empty.
+	// Empty list means compatible to all providers.
```

Comment:
```suggestion
	// CompatibleProviders is a list of provider names which this app is compatible with. Default is empty.
	// Empty list means app is compatible with all providers.
```

Let's make it `Default to` to align with the other fields.